### PR TITLE
[bugfix] Catch "fatal: Not a git repository (or any of the parent directories): .git" exception at `GitHelper::isGitDir()`

### PR DESCRIPTION
**Pull request by phansys**
|Q            |A   |
|---          |--- |
|Fixed tickets|#612|
|License      |MIT |

### CODE REVIEW

File: src/Helper/GitHelper.php [hunk 1]
```diff
@@ -140,7 +142,15 @@ public function restoreStashedBranch()
      */
     public function isGitDir($requireRoot = true)
     {
-        $directory = $this->getGitDir();
+        try {
+            $directory = $this->getGitDir();
+        } catch (\RuntimeException $e) {
+            if (128 === $e->getCode() && 'fatal: Not a git repository (or any of the parent directories): .git' === $e->getMessage()) {
```
Comment (GrahamCampbell) [inline src/Helper/GitHelper.php]: Maybe it's fine if this starts with `fatal: Not a git repository` with anything else after that?

Comment (GrahamCampbell) [inline src/Helper/GitHelper.php]: Might be cool to define 128 as a constant somewhere?

File: src/Helper/ProcessHelper.php [hunk 2]
```diff
@@ -78,7 +78,7 @@ public function runCommand($command, $allowFailures = false, $callback = null)
         $process->run($callback);
 
         if (!$process->isSuccessful() && !$allowFailures) {
-            throw new \RuntimeException($process->getErrorOutput());
+            throw new \RuntimeException(trim($process->getErrorOutput()), $process->getExitCode());
```
Comment (GrahamCampbell) [inline src/Helper/ProcessHelper.php]: Maybe throw something more specific that extends runtime exception

File: src/Helper/GitHelper.php (hunk 1 above)
Comment (phansys) [inline src/Helper/GitHelper.php]: That was my first thought @GrahamCampbell, but I don't know which is the git's mapping for exit codes.

File: src/Helper/ProcessHelper.php (hunk 2 above)
Comment (phansys) [inline src/Helper/ProcessHelper.php]: There isn't any exception suitable for this AFAIK. Current exceptions extending directly `\RuntimeException` are:
* `AdapterException`
* `FileNotFoundException`
* `InvalidStateException`
* `NotImplementedException`
* `UnknownRemoteException`
* `UnsupportedTypeException`

**Comment by cordoval** [comment]
can we leave this unhandled and let the exception tell us what is wrong rather than doing this?

File: src/Helper/GitHelper.php (hunk 1 above)
Comment (GrahamCampbell) [inline src/Helper/GitHelper.php]: I was meaning we just define our own constant.

File: src/Helper/ProcessHelper.php (hunk 2 above)
Comment (GrahamCampbell) [inline src/Helper/ProcessHelper.php]: Add one perhaps?

**Comment by GrahamCampbell** [comment]
> can we leave this unhandled and let the exception tell us what is wrong rather than doing this?

That could be a good idea too. Hmmm.

**Comment by cordoval** [comment]
yes, let's remove the test which is more of the problem

**Pull request closed without merging**